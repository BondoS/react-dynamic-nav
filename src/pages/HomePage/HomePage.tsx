import React from 'react';
import { Aside } from '../../components/Aside';
import { Header } from '../../components/Header';
import { HeaderNav } from '../../components/HeaderNav';
import { LeftNav } from '../../components/LeftNav';
import { Main } from '../../components/Main';
import { MidContent } from '../../components/MidContent';

type Props = {};

export const HomePage = ({}: Props) => {
  return (
    <>
      <Aside>
        <LeftNav>left nav</LeftNav>
      </Aside>
      <MidContent>
        <Header>
          <HeaderNav>Header Nav</HeaderNav>
        </Header>
        <Main>Main</Main>
      </MidContent>
    </>
  );
};
