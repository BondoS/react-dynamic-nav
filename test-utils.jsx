import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { globalReducer, initialGlobalState } from './src/store';

// A function that I normally make available for all my tests
// that deal with connected components.

// We can provide initialState or the entire store that the ui is rendered with
export const render = (
  ui,
  {
    initialState = initialGlobalState,
    store = configureStore({
      reducer: globalReducer,
      middleware: (getDefaultMiddleware) => [...getDefaultMiddleware()],
    }),
    ...renderOptions
  } = {}
) => {
  const Wrapper = ({ children }) => (
    <Provider store={store}>{children}</Provider>
  );

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};
