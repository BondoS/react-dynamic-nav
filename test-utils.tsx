import { ReactElement } from 'react';
import { render as rtlRender, RenderOptions } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { globalReducer } from './src/store';
import logger from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import rootSaga from './src/store/sagas';

const sagaMiddleware = createSagaMiddleware();

const store = configureStore({
  reducer: globalReducer,
  middleware: (getDefaultMiddleware) => [
    ...getDefaultMiddleware({ thunk: false }).concat(logger),
    sagaMiddleware,
  ],
});
sagaMiddleware.run(rootSaga);

const render = (ui: ReactElement, renderOptions?: RenderOptions) => {
  const Wrapper: React.FC = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

export { render };
