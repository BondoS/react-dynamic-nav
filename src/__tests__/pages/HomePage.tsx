import { waitFor, screen } from '@testing-library/react';
import { render } from '../../../test-utils';
import { HomePage } from '../../pages/HomePage';

jest.mock('../../services/getNavListService', () => {
  return {
    getNavListService: jest.fn().mockImplementation(() =>
      Promise.resolve([
        {
          title: 'Mocked Reporting',
          url: null,
          id: 'menu-reporting',
          icon: 'reporting',
          hasAlert: false,
          children: [],
        },
      ])
    ),
  };
});

test('Calls endpoint and render nav item', async () => {
  const view = render(<HomePage />);
  expect(view).toMatchSnapshot();
  await waitFor(() => {
    expect(screen.getAllByTestId('itemHead').length).toBeGreaterThanOrEqual(1);
  });
});
