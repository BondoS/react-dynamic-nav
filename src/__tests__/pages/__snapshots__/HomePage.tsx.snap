// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calls endpoint and render nav item 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <aside
        class="sc-bczRLJ fjwDpE"
        open=""
      >
        <button
          aria-expanded="true"
          class="toggleDrawerBtn"
          type="button"
        >
          <img
            alt="Toggle left drawer"
            class="toggleDrawerImg"
            src="list.svg"
          />
        </button>
        <nav
          class="sc-gKXOVf iHJRWE"
        >
          <ul>
            <li
              class="sc-eCYdqJ emajkZ closed"
              id="menu-reporting"
            >
              <button
                aria-expanded="false"
                class="itemHead"
                data-testid="itemHead"
              >
                <div
                  class="logoAndTitle"
                >
                  <img
                    alt="nav logo"
                    class="itemLogo"
                    src="basket3.svg"
                  />
                  <span
                    class="titleText"
                  >
                    Mocked Reporting
                  </span>
                </div>
              </button>
              <div
                class="itemBody"
              />
            </li>
          </ul>
        </nav>
      </aside>
      <div
        class="sc-ftvSup"
      >
        <header
          class="sc-hKMtZM eKgZKc"
        >
          <img
            alt="logo"
            class="sc-gsnTZi exkqoG"
            src="wayfair-partner-home-logo.svg"
          />
          <nav
            class="sc-dkzDqf"
          />
        </header>
        <main
          class="sc-iBkjds"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <aside
      class="sc-bczRLJ fjwDpE"
      open=""
    >
      <button
        aria-expanded="true"
        class="toggleDrawerBtn"
        type="button"
      >
        <img
          alt="Toggle left drawer"
          class="toggleDrawerImg"
          src="list.svg"
        />
      </button>
      <nav
        class="sc-gKXOVf iHJRWE"
      >
        <ul>
          <li
            class="sc-eCYdqJ emajkZ closed"
            id="menu-reporting"
          >
            <button
              aria-expanded="false"
              class="itemHead"
              data-testid="itemHead"
            >
              <div
                class="logoAndTitle"
              >
                <img
                  alt="nav logo"
                  class="itemLogo"
                  src="basket3.svg"
                />
                <span
                  class="titleText"
                >
                  Mocked Reporting
                </span>
              </div>
            </button>
            <div
              class="itemBody"
            />
          </li>
        </ul>
      </nav>
    </aside>
    <div
      class="sc-ftvSup"
    >
      <header
        class="sc-hKMtZM eKgZKc"
      >
        <img
          alt="logo"
          class="sc-gsnTZi exkqoG"
          src="wayfair-partner-home-logo.svg"
        />
        <nav
          class="sc-dkzDqf"
        />
      </header>
      <main
        class="sc-iBkjds"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
